pkg/karmadactl/promote/promote.go

```diff
@@ -393,41 +518,41 @@ func (o *CommandPromoteOption) printObjectAndPolicy(obj *unstructured.Unstructur
 }
 
 // createPropagationPolicy create PropagationPolicy in karmada control plane
-func (o *CommandPromoteOption) createPropagationPolicy(karmadaClient *karmadaclientset.Clientset, gvr schema.GroupVersionResource) error {
+func (o *CommandPromoteOption) createPropagationPolicy(karmadaClient *karmadaclientset.Clientset, obj *unstructured.Unstructured, gvr schema.GroupVersionResource, isDep bool) error {
 	var policyName string
-	if o.PolicyName == "" {
-		policyName = names.GeneratePolicyName(o.Namespace, o.name, o.gvk.String())
+	if o.PolicyName == "" || isDep {
+		policyName = names.GeneratePolicyName(obj.GetNamespace(), obj.GetName(), obj.GroupVersionKind().String())
 	} else {
 		policyName = o.PolicyName
 	}
 
-	_, err := karmadaClient.PolicyV1alpha1().PropagationPolicies(o.Namespace).Get(context.TODO(), policyName, metav1.GetOptions{})
+	_, err := karmadaClient.PolicyV1alpha1().PropagationPolicies(obj.GetNamespace()).Get(context.TODO(), policyName, metav1.GetOptions{})
 	if err != nil && apierrors.IsNotFound(err) {
-		pp := buildPropagationPolicy(o.name, policyName, o.Namespace, o.Cluster, gvr, o.gvk)
-		_, err = karmadaClient.PolicyV1alpha1().PropagationPolicies(o.Namespace).Create(context.TODO(), pp, metav1.CreateOptions{})
+		pp := buildPropagationPolicy(obj.GetName(), policyName, obj.GetNamespace(), o.Cluster, gvr, obj.GroupVersionKind())
+		_, err = karmadaClient.PolicyV1alpha1().PropagationPolicies(obj.GetNamespace()).Create(context.TODO(), pp, metav1.CreateOptions{})
 
 		return err
 	}
 	if err != nil {
-		return fmt.Errorf("failed to get PropagationPolicy(%s/%s) in control plane: %v", o.Namespace, policyName, err)
+		return fmt.Errorf("failed to get PropagationPolicy(%s/%s) in control plane: %v", obj.GetNamespace(), policyName, err)
 	}
 
 	// PropagationPolicy already exists, not to create it
-	return fmt.Errorf("the PropagationPolicy(%s/%s) already exist, please edit it to propagate resource", o.Namespace, policyName)
+	return fmt.Errorf("the PropagationPolicy(%s/%s) already exist, please edit it to propagate resource", obj.GetNamespace(), policyName)
 }
 
 // createClusterPropagationPolicy create ClusterPropagationPolicy in karmada control plane
-func (o *CommandPromoteOption) createClusterPropagationPolicy(karmadaClient *karmadaclientset.Clientset, gvr schema.GroupVersionResource) error {
+func (o *CommandPromoteOption) createClusterPropagationPolicy(karmadaClient *karmadaclientset.Clientset, obj *unstructured.Unstructured, gvr schema.GroupVersionResource, isDep bool) error {
 	var policyName string
-	if o.PolicyName == "" {
-		policyName = names.GeneratePolicyName("", o.name, o.gvk.String())
+	if o.PolicyName == "" || isDep {
+		policyName = names.GeneratePolicyName("", obj.GetName(), obj.GroupVersionKind().String())
```

Comment:
Do we really need to create the dependencies' PropagationPolicy?
How about just creating the resource and enabling the `propagateDeps` in the main resources' PP.
For example:
I want to promote a deployment, and it asks for one configmap. So just
1. create configmap/deployment in karmada.
2. Create PP(`propagateDeps` enabled) for  deployment.

Referring to: https://karmada.io/zh/docs/userguide/scheduling/propagate-dependencies/

Reply:
cc @chaunceyjiang @XiShanYongYe-Chang  for suggestions

Reply:
This is a bit similar to the MCS feature we're [working on](https://github.com/karmada-io/karmada/pull/3769), which also uses follow distribution. It might not be a good idea to create the pp directly as it is a user-oriented resource, is it possible to try to create the rb resource.